mirrord/agent/src/steal/ip_tables/chain.rs

```diff
@@ -0,0 +1,92 @@
+use std::sync::{
+    atomic::{AtomicI32, Ordering},
+    Arc,
+};
+
+use crate::{
+    error::{AgentError, Result},
+    steal::ip_tables::IPTables,
+};
+
+#[derive(Debug)]
+pub struct IPTableChain<IPT: IPTables> {
+    inner: Arc<IPT>,
+    chain: String,
+    chain_size: AtomicI32,
+}
+
+impl<IPT> IPTableChain<IPT>
+where
+    IPT: IPTables,
+{
+    pub fn create(inner: Arc<IPT>, chain: String) -> Result<Self> {
+        inner.create_chain(&chain)?;
+
+        // Start with 1 because the chain will allways have atleast `-A <chain name>` as a rule
+        let chain_size = AtomicI32::from(1);
+
+        Ok(IPTableChain {
+            inner,
+            chain,
+            chain_size,
+        })
+    }
+
+    pub fn load(inner: Arc<IPT>, chain: String) -> Result<Self> {
+        let existing_rules = inner.list_rules(&chain)?.len();
+
+        if existing_rules == 0 {
+            return Err(AgentError::IPTablesError(format!(
+                "Unable to load rules for chain {chain}"
+            )));
+        }
+
+        // Start with 1 because the chain will allways have atleast `-A <chain name>` as a rule
+        let chain_size = AtomicI32::from((existing_rules - 1) as i32);
+
+        Ok(IPTableChain {
+            inner,
+            chain,
+            chain_size,
+        })
+    }
+
+    pub fn chain_name(&self) -> &str {
```

Comment:
Should be named the same as `chain` (like, either both should be named `chain` or both should be named `chain_name`